rust/src/treefile.rs

```diff
@@ -1279,6 +1279,15 @@ pub(crate) struct DeriveConfigFields {
 }
 
 impl DeriveConfigFields {
+    /// Returns `true` if there are fields set which override the content.
+    pub(crate) fn has_overrides(&self) -> bool {
```

Comment:
Shouldn't this basically just duplicate all of `rpmostree_origin_may_require_local_assembly`?

Reply:
Yeah, I was alluding to that in the commit message:

>  Ultimately, the code duplication here will go away when we switch to converting an origin to a treefile on reading, and always write treefiles.

I guess one thing we could try here is to generate a treefile in `rpmostree-origin.cxx` and then have our methods be based on that?

Reply:
So IIUC, right now we'll still be emitting warnings if e.g. we use `initramfs-etc`, right?

Hmm, I guess a more incremental step we could do is to move the logic of `rpmostree_origin_may_require_local_assembly` to Rust where it converts to a treefile first and then queries the same fields (that part split in an `_inner` function or whatever). Then we could have `treefile_to_origin_inner` just use the `_inner` function instead?

Reply:
> So IIUC, right now we'll still be emitting warnings if e.g. we use initramfs-etc, right?

Yeah.

> Hmm, I guess a more incremental step we could do is to move the logic of rpmostree_origin_may_require_local_assembly to Rust

Agree - but this is a case of asking for a bigger change to fix more bugs when we have a smaller change to fix fewer bugs (and low risk of new bugs).  Let's get this in and keep that as a followup?

Reply:
Actually the smallest and lowest risk thing is probably to just drop the warning for now.

Reply:
> Actually the smallest and lowest risk thing is probably to just drop the warning for now.

Right, I think that's a better fix. I hadn't realized we were currently tripping warnings. This isn't really something for users to care about at this early point in the process.

Reply:
https://github.com/coreos/rpm-ostree/pull/2975